File: LENGUAJES/PYTHON/Ejercicio5.py
#Condicionales IF, ELSE

def obtener_edad():
    while True:
        try:
            edad = int(input("Ingresa tu edad: "))
            if edad < 0:
                print("La edad no puede ser negativa. Inténtalo de nuevo.")
            else:
                return edad
        except ValueError:
            print("Error, por favor ingresa un número entero válido.")

def mayoria_edad():
    edad = obtener_edad()
    
    if edad >= 18:
        print("Eres mayor de edad.")
    else:
        print("Eres menor de edad.")
mayoria_edad()

File: LENGUAJES/PHP/Ejercicio5.php
<?php
//CONDICIONALES

function obtenerEdad() {
    while (true) {
        echo "Ingresa tu edad: ";
        $input = trim(fgets(STDIN));
        
        if (is_numeric($input) && intval($input) == $input && $input >= 0) {
            return intval($input);
        } else {
            echo "Error, por favor ingresa un número entero válido y no negativo.\n";
        }
    }
}

function mayoriaEdad() {
    $edad = obtenerEdad();
    
    if ($edad >= 18) {
        echo "Eres mayor de edad.\n";
    } else {
        echo "Eres menor de edad.\n";
    }
}
mayoriaEdad();
?>
